data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -83,25 +81,22 @@ public GetForkResponse getForkInfo() {
     if (!combinedChainDataClient.isChainDataFullyAvailable()) {
       return chainUnavailable();
     }
-    final UnsignedLong committeesCalculatedAtEpoch = epoch.equals(ZERO) ? ZERO : epoch.minus(ONE);
-    final UnsignedLong startingSlot = compute_start_slot_at_epoch(committeesCalculatedAtEpoch);
-    final UnsignedLong slot = compute_start_slot_at_epoch(epoch);
-
-    // one epoch in future is available, beyond that cannot be calculated
-    if (slot.compareTo(
-            recentChainData.getBestSlot().plus(UnsignedLong.valueOf(Constants.SLOTS_PER_EPOCH)))
-        > 0) {
+
+    final UnsignedLong earliestQueryableSlot =
+        CommitteeUtil.getEarliestQueryableSlotForTargetEpoch(epoch);
+    if (recentChainData.getBestSlot().compareTo(earliestQueryableSlot) < 0) {
```

Comment:
This check means that if we have a span of time where no blocks are produced, we won't be able to query committee information for epochs ahead of the epoch after the latest block (from the REST API).  Is this right?

Reply:
I think we now progress the best slot even if the slots are empty so this should work out ok.  Worth testing to be sure though.

Reply:
The slot we pull is the slot from the head block, not the latest slot processed.  I think what we probably want to do here is return an empty result if we're syncing, or return an error if the slot is from the future. 🤔

Reply:
Hmm, you'er right but we do track the actual best slot - `updateBestBlock` is called with the nodeSlot even if the slot was empty to ensure we treat It as a reorg if we later import blocks at those slots.  So if `getBestSlot` (or possibly a new method) returned `chainHead.getSlot()` we'd get the current slot regardless of any empty slots.